File: frontend/src/services/commentService.js
import api from './api.jsx';

export const getCommentsByTicket = async (ticketId, page = 0, size = 10) => {
    const response = await api.get(`/comments/ticket/${ticketId}?page=${page}&size=${size}`);
    return response.data;
};

export const addComment = async (ticketId, text) => {
    const response = await api.post(`/comments/ticket/${ticketId}`, { text });
    return response.data;
};

export const updateComment = async (commentId, text) => {
    const response = await api.put(`/comments/${commentId}`, { text });
    return response.data;
};

export const deleteComment = async (commentId) => {
    await api.delete(`/comments/${commentId}`);
};

File: backend/src/java/com/smartcampus/service/interfaces/CommentService.java
package com.smartcampus.service.interfaces;

import com.smartcampus.model.dto.CommentDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface CommentService {
    CommentDTO addComment(Long ticketId, String text, String userEmail);
    Page<CommentDTO> getCommentsByTicket(Long ticketId, Pageable pageable);
    CommentDTO updateComment(Long commentId, String newText, String userEmail);
    void deleteComment(Long commentId, String userEmail);
}